src/server/scripts/EasternKingdoms/zone_goldshire.cpp

```diff
@@ -0,0 +1,271 @@
+    {
+        switch (pathId)
+        {
+            case STORMWIND_PATH:
+            {
+                if (waypointId == STORMWIND_WAYPOINT)
+                {
+                    me->GetMotionMaster()->MoveRandom(10.f);
+                    _events.ScheduleEvent(EVENT_WP_START_GOLDSHIRE, 11min);
+                }
+
+                break;
+            }
+            case GOLDSHIRE_PATH:
+            {
+                if (waypointId == GOLDSHIRE_WAYPOINT)
+                {
+                    me->GetMotionMaster()->MoveRandom(10.f);
+                    _events.ScheduleEvent(EVENT_WP_START_WOODS, 15min);
+                }
+                break;
+            }
+            case WOODS_PATH:
+            {
+                if (waypointId == WOODS_WAYPOINT)
+                {
+                    me->GetMotionMaster()->MoveRandom(10.f);
+                    _events.ScheduleEvent(EVENT_WP_START_HOUSE, 6min);
+                    _events.ScheduleEvent(EVENT_WP_START_LISA, 2s);
+                }
+
+                break;
+            }
+            case HOUSE_PATH:
+            {
+                if (waypointId == HOUSE_WAYPOINT)
+                {
+                    // Move childeren at last point 
+                    MoveTheChildren();
+
+                    // After 30 seconds a random sound should play
+                    _events.ScheduleEvent(EVENT_PLAY_SOUNDS, 30s);
+                }
+                break;
+            }
+        }
+    }
+
+    void OnGameEvent(bool start, uint16 eventId) override
+    {
+        if (start && eventId == GAME_EVENT_CHILDEREN_OF_GOLDSHIRE)
+        {
+            // Start event at 7 am
+            // Begin pathing
+            _events.ScheduleEvent(EVENT_BEGIN_EVENT, 2s);
+            _started = true;
+        }
+        else if (!start && eventId == GAME_EVENT_CHILDEREN_OF_GOLDSHIRE)
+        {
+            // Reset event at 8 am
+            _started = false;
+            _events.Reset();
+        }
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        if (!_started)
+            return;
+
+        _events.Update(diff);
+
+        while (uint32 eventId = _events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+                case EVENT_WP_START_GOLDSHIRE:
+                    me->GetMotionMaster()->MovePath(GOLDSHIRE_PATH, false);
+                    break;
+                case EVENT_WP_START_WOODS:
+                    me->GetMotionMaster()->MovePath(WOODS_PATH, false);
+                    break;
+                case EVENT_WP_START_HOUSE:
+                    me->GetMotionMaster()->MovePath(HOUSE_PATH, false);
+                    break;
+                case EVENT_WP_START_LISA:
+                    for (uint32 i = 0; i < _childrenGUIDs.size(); ++i)
+                        if (Creature* lisa = ObjectAccessor::GetCreature(*me, _childrenGUIDs[i]))
+                            if (lisa->GetEntry() == NPC_LISA)
```

Comment:
I think you could call "_childrenGUIDs[i].GetEntry() == NPC_LISA" before GetCreature() to check if the GUID is of the NPC entry you are looking for

Reply:
nvm, it's ok to leave the GetEntry() check